climada/engine/forecast.py

```diff
@@ -0,0 +1,897 @@
+        x_ticks = list()
+        ticklabel_dict = {pow(10, i): self._number_to_str(pow(10, i)) for i in range(0, 15)} #turn int to str
+
+        for i in np.arange(15):
+            tick_i = 10.0 ** i
+            x_ticks.append(tick_i)
+            x_ticklabels.append(ticklabel_dict[tick_i])
+        ax.xaxis.set_ticks(x_ticks)
+        ax.xaxis.set_ticklabels(x_ticklabels)
+        plt.xticks(rotation=15, horizontalalignment='right')
+        plt.xlim([(10 ** -0.25) * bins[0], (10 ** 0.25) * bins[-1]])
+
+        lead_time_str = '{:.0f}'.format(self.lead_time(run_datetime).days + self.lead_time(run_datetime).seconds/60/60/24) # cant show '2' and '2.5'
+        title_dict = {'event_day': self.event_date.strftime('%a %d %b %Y 00-24UTC'),
+                      'run_start': run_datetime.strftime('%d.%m.%Y %HUTC +') + lead_time_str + 'd',
+                      'explain_text': ('total building damage caused by wind'),
+                      'model_text': "CLIMADA IMPACT"}
+        title_position = {'model_text': [0.13, 0.94], 'explain_text': [0.13, 0.9], 'event_day': [0.9, 0.94],
+                          'run_start': [0.9, 0.9]}
+        left_right = {'model_text': 'left', 'explain_text': 'left', 'event_day': 'right', 'run_start': 'right'}
+        color = {'model_text': 'k', 'explain_text': 'r', 'event_day': 'k', 'run_start': 'k'}
+        for t in title_dict:
+            plt.figtext(title_position[t][0],
+                        title_position[t][1],
+                        title_dict[t],
+                        fontsize='xx-large',
+                        color=color[t],
+                        ha=left_right[t])
+
+        plt.xlabel('forecasted total damage for ' + self.exposure_name +
+                   ' [' + self._impact[haz_ind].unit + ']')
+        plt.ylabel('probability')
+        plt.text(0.75, 0.85,
+                 # 'mean damage:\n' + locale.currency(np.round(imp.at_event.mean()), grouping=True),
+                 'mean damage:\nCHF ' + '{0:,}'.format(np.int(np.round(self._impact[haz_ind].at_event.mean()))).replace(',', "'"),
+                 horizontalalignment='center',
+                 verticalalignment='center',
+                 transform=ax.transAxes)
+        if save_fig:
+            plt.savefig(histbin_file_name_full)
+        if close_fig:
+            plt.clf()
+            plt.close(f)
+
+    @staticmethod
+    def _number_to_str(number):
+        """ Generate numbers as str (with thousand, million, trillion).
+            Example: 1000 becomes 1 thousand
+
+        Parameters
+        ----------
+        number: int
+
+        Returns
+        -------
+        str
+        """
+        number_names = ['', ' thousand', ' million', ' billion', ' trillion']
+        millidx = max(0, min(len(number_names) - 1,
+                             int(math.floor(0 if number == 0 else math.log10(abs(number)) / 3))))
+        return '{:.0f}{}'.format(number / 10 ** (3 * millidx), number_names[millidx])
```

Comment:
This functionality already exists `util.value_repr.value_to_monetary_unit()`. Please use this instead.